dags/glam.py

```diff
@@ -306,22 +262,20 @@
 clients_scalar_aggregates >> clients_scalar_bucket_counts
 clients_scalar_aggregates >> scalar_percentiles
 
-latest_versions >> clients_daily_histogram_aggregates
+latest_versions >> glam_valid_builds
```

Comment:
Initially I put `glam_valid_builds` right after `clients_histogram_aggregates` because that is what it depends on (aka it queries from that table) but you also mentioned that `clients_daily_histogram_aggregates` depends on `glam_valid_build` so it should come before that. I obviously don't want to make a cycle in the graph, so I want to make sure we're on the same page for this.

If I'm understanding correctly, you mean that we could do 

`wait_for_main_ping` -> `glam_valid_builds`
`wait_for_main_ping` -> `latest_versions`
`glam_valid_builds` -> `clients_daily_histogram_aggregates`
`latest_versions` -> `clients_daily_histogram_aggregates`

Is this what you mean?

Reply:
Yep, that's correct.

On a slightly different thread, will `glam_valid_builds` pick up on the new valid builds if it comes before `clients_daily_histogram_aggregates`?

Reply:
Hmmmm now that I think of it, maybe this whole approach doesn't work.... Because a build ID can only become valid after it has built up enough clients in it but if it's invalid it won't be included and can't build up clients.

In that case, we can't take advantage of filtering invalid clients early on, and we can just do the filtering after `clients_histogram_aggregates` has been computed so that it doesn't make it to the front end.

@acmiyaguchi what do you think?

Reply:
That sounds right to me. It's fine to push the filtering out to the end for cosmetic reasons (and hopefully the buildhub2 dataset can be a stand in replacement), but the original idea behind filtering was to reduce join complexity.

I'm not clear on what the state of the DAG is currently -- is it running successfully without the changes here to increase the number of partitions?

Reply:
> but the original idea behind filtering was to reduce join complexity.

Yes, I mentioned here in agreement with you: https://github.com/mozilla/bigquery-etl/pull/754#issuecomment-588501341 that the build ID filtering would just be for helping the UI at this point and wouldn't really help with the join complexity. 

> I'm not clear on what the state of the DAG is currently -- is it running successfully without the changes here to increase the number of partitions?

Yes, the dag has been running successfully for 10 days without these changes. It's a little slow though, so it would stand to benefit from the parallelization done here. I think I want to get the non-build_id changes out ASAP to speed things up so I'm going to split the build ID stuff off from here. I'm not even sure there is much benefit to implementing this approach then soon after switching to build hub.

Reply:
Okay, it's good to hear that these changes aren't blocking successful runs. The parallelization may not help all that much because each project shares limited resources (e.g. BigQuery slots or equivalent limit). We've had luck running large-scale backfills in separate projects like `moz-fx-data-backfill-{1..31}`, though. Airflow will also limit the parallelism in a worker pool, so it may be worth checking out the landing times/gantt chart if this change goes out.

Reply:
It's mostly the other jobs that haven't been broken up yet that are slow + at risk for shuffle failures - `clients_histogram_aggregates_final` and `clients_histogram_aggregates_old`. I think in this case, it would be best to start with 4 partitions like I've done for `clients_histogram_aggregates_merged` and see how it plays out.